### PR TITLE
Store records for incidents on activating

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Handles storing the record for migrated processors. Currently incident handling need to have the record in the state, this will be removed later.

I created a separate PR because we need that in other processors as well, so we can make better progress on others as well.

## Related issues

related #6195 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Review by korthout — APPROVED** [review]
LGTM 🐶

**Comment by korthout** [comment]
bors merge

**Comment by Zelldon** [comment]
I will take a look @korthout

File: engine/src/main/java/io/zeebe/engine/processing/job/JobCompleteProcessor.java [hunk 1]
```diff
@@ -73,9 +67,26 @@ public void afterAccept(
...
         // for COMPLETE_ELEMENT commands; switch out for command writer
-        stateWriter.appendFollowUpEvent(
-            serviceTaskKey, WorkflowInstanceIntent.ELEMENT_COMPLETING, workflowInstanceRecord);
+        if (MigratedStreamProcessors.isMigrated(BpmnElementType.SERVICE_TASK)) {
+          stateWriter.appendFollowUpEvent(
+              serviceTaskKey, WorkflowInstanceIntent.ELEMENT_COMPLETING, workflowInstanceRecord);
+        } else {
+          eventWriter.appendFollowUpEvent(
+              serviceTaskKey, WorkflowInstanceIntent.ELEMENT_COMPLETING, workflowInstanceRecord);
+        }
```
Comment (Zelldon) [inline engine/src/main/java/io/zeebe/engine/processing/job/JobCompleteProcessor.java:76]: I think it is enough to just have `eventWriter.appendFollowUpEvent(
              serviceTaskKey, WorkflowInstanceIntent.ELEMENT_COMPLETING, workflowInstanceRecord);` we should migrate that if we migrate service task.

**Review by ChrisKujawa — COMMENTED** [review]
Thanks @korthout just one thing

File: engine/src/main/java/io/zeebe/engine/processing/job/JobCompleteProcessor.java (hunk 1 above)
Comment (korthout) [reply engine/src/main/java/io/zeebe/engine/processing/job/JobCompleteProcessor.java:76]: I would leave it, this allows for easier migration of the service task, and this feature toggle will be removed in #6202

**Comment by korthout** [comment]
bors merge

**Pull request merged**